scripts/changePartnerShares.ts

```diff
@@ -0,0 +1,45 @@
+import { BigNumberish } from 'ethers';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { ICollector } from 'typechain-types/contracts/Collector';
+import { parseJsonFile } from './utils';
+
+export type ChangePartnerSharesArg = {
+  collectorAddress: string;
+  partnerConfig: string;
+  gasLimit?: BigNumberish;
+};
+
+export type PartnerConfig = {
+  partners: ICollector.RevenuePartnerStruct[];
+};
+
+const defaultTxGas = 150000;
+
+export const changePartnerShares = async (
+  {
+    collectorAddress,
+    partnerConfig,
+    gasLimit = defaultTxGas,
+  }: ChangePartnerSharesArg,
+  hre: HardhatRuntimeEnvironment
+) => {
+  const { ethers } = hre;
+
+  const parsedPartnerConfig = parseJsonFile<PartnerConfig>(
+    partnerConfig
+  ) as PartnerConfig;
```

Comment:
We are forcing the type here since we are assuming that it will be failing instead of returning an undefined value, right?

Reply:
We don't need it actually, thank you for pointing it out.